src/include/logging/wal_log_manager.h

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// wal_log_manager.h
+//
+// Identification: src/logging/wal_log_manager.h
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+#include <fstream>
+
+namespace peloton{
+namespace logging{
+
+class LogManager{
+public:
+
+  LogManager() {}
+  ~LogManager() {}
+
+  inline bool init(std::string logging_dir) {
+    directory_ = logging_dir;
+
+    if(!logger_ofstream_.is_open()){
+      logger_ofstream_.open(logging_dir+"/log", std::ofstream::out | std::ofstream::app);
+
+      if(!logger_ofstream_.fail()) {
+        enable_logging_ = true;
+        return true;
+      }
+
+    }
+
+    enable_logging_ = false;
+    return false;
+  }
+
+  inline static LogManager &GetInstance(){
+    static LogManager log_manager;
+    return log_manager;
+  }
+
+  inline static size_t GetTransactionBufferSize() { return transaction_buffer_threshold_; }
+  inline static size_t GetLoggerBufferSize() { return logger_buffer_threshold_; }
+
+  inline bool IsLoggingEnabled() { return enable_logging_; }
+  inline std::string GetDirectory() { return directory_; }
+  inline std::string GetLogFilePath() { return directory_+"/tmp"; }
+  inline std::ofstream *GetFileStream() { return &logger_ofstream_; }
+
+//  TODO(gandeevan): support StartLogging and StopLogging
+//  static void StartLogging() {}
+//  static void StopLogging() {}
+
+
+private:
+
+  const static size_t logger_buffer_threshold_ = 1024 * 512;  // 512 KB
```

Comment:
I think these could be declared with other database toggles?

Reply:
@Onmysofa is correct. We should move this into `pg_settings`

Reply:
Yes, we will move it as well as some others into `pg_settings` in the end.

Reply:
Done